astropy/nddata/mixins/ndslicing.py

```diff
@@ -30,7 +30,7 @@ class NDSlicingMixin:
 
         >>> nd = NDDataSliceable([1,2,3,4,5])
         >>> nd[1:3]
-        NDDataSliceable([2, 3])
+        NDDataSliceable([2 3])
```

Comment:
That's actually a bit confusing, since the `__repr__` of `Masked` does have `,`; it follows what `ndarary` (and `np.ma.MaskedArray`) does:
```
In [10]: str(Masked(np.array([1, 2.])))
Out[10]: '[1. 2.]'

In [11]: repr(Masked(np.array([1, 2.])))
Out[11]: 'MaskedNDArray([1., 2.])'

In [12]: str(np.array([1, 2.]))
Out[12]: '[1. 2.]'

In [13]: repr(np.array([1, 2.]))
Out[13]: 'array([1., 2.])'
```

I agree that here it should have`,` too -- maybe have a look at what I did for `MaskedNDArray`?